### DISCUSSION TITLE
--test always shows no tests available

### DISCUSSION THREAD

**Discussion by the-emerald** [pacak/cargo-show-asm#130] [Q&A] [answered]
Try the following:
```
cargo new whatever --lib
cd whatever
```
The default library template comes with a test called `tests::it_works`.
```
cargo asm --test
```
shows "No tests available." Trying to specify the test:
```
cargo asm --test tests::it_works
```
doesn't work either, as it shows "no test target named `tests::it_works`."

**Comment by pacak** [comment, marked as answer]
`--test` in `cargo-show-asm` behaves the same way `--test` works in `cargo build` - it builds integration test. To look at the assembly (llvm, mir, etc) of the test themselves you need to build a library with the test profile: `cargo asm --profile test it_works 0` - this works for default library template for example.

**Comment by the-emerald** [comment]
Ah, I see! I thought the `--test` was for library tests. Looking at #1 I can now see that it's for tests in a `test/` directory. Thanks!
